app/helpers/locales_helper.rb

```diff
@@ -1,74 +1,80 @@
-#require 'http_accept_language'
-
 module LocalesHelper
-
-  def process_locale
-    I18n.locale = best_locale
+  def use_preferred_locale
+    I18n.locale = preferred_locale
     yield if block_given?
     save_detected_locale
   end
 
-  def selectable_locales
-    Loomio::I18n::SELECTABLE_LOCALES
+  def preferred_locale
+    strict_filter(params[:locale]) ||
+    fuzzy_filter(user_selected_locale) ||
+    fuzzy_filter(browser_detected_locales) ||
+    fuzzy_filter(user_detected_locale) ||
+    I18n.default_locale
   end
 
-  def detectable_locales
-    Loomio::I18n::DETECTABLE_LOCALES
-  end
-
-  def rtl_locales
-    Loomio::I18n::RTL_LOCALES
+  def supported_locales
+    YAML.load_file(Rails.root.join('config', 'locales.yml'))["supported_locales"]
```

Comment:
I'd recommend throwing this into AppConfig, like this:

```
# app_config.rb
CONFIG_FILE = %w(... locales)
```
then
```
AppConfig.locales['supported_locales']
```

which means we likely don't need this method either.